### PR TITLE
add support for DRO v2

### PR DESCRIPTION
**Pull request by rofl0r**
unfortunately some whitespace changes had to be done because the codeflow was almost unreadable due to mixed tabs/spaces.
adding ?w=1 to the end of the file changes url makes them disappear for review.

### CODE REVIEW

File: dro2midi.cpp [hunk 1]
```diff
@@ -946,6 +956,15 @@ void doNoteOnOff(bool bKeyOn, int chanOPL, int chanMIDI)
 	return;
 }
 
+static const char* dro2hwtypestr(unsigned hwtype) {
+	switch(hwtype) {
+	case 0: return "OPL2";
+	case 1: return "OPL2 dual";
+	case 2: return "OPL3";
+	default: return "UNKWNOWN";
```
Comment (Malvineous) [inline dro2midi.cpp]: Typo

File: droshrink.c [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+	dro2hdr.iLengthMS = readUINT32LE(in);
+	fread(&dro2hdr.iHardwareType, 1, 6, in);
+	if(dro2hdr.iCodemapLength >= 128) {
+		fprintf(stderr, "invalid setting %u for iCodemapLength!\n", (unsigned) dro2hdr.iCodemapLength);
+		return 2;
+	}
+	fread(dro2hdr.iCodemap, 1, dro2hdr.iCodemapLength, in);
+	dro2hdr_out = dro2hdr;
+	dro2hdr_out.iLengthPairs -= n;
+	dro2hdr_out.iLengthMS -= n; /* FIXME: how to accurately fixup time? */
```
Comment (Malvineous) [inline droshrink.c]: Just run through all the events and total the delay values, and put the total number here.  Or if you are removing events, you can sum the delays on all the removed events and then subtract that from the original value.

**Review by Malvineous — CHANGES_REQUESTED** [review]
Looks good, thanks for the PR!  Will merge if you can fix the typo, up to you about the other comment.

**Comment by rofl0r** [comment]
alright, updated. good idea with the delay counting.

File: droshrink.c [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+	dro2hdr.iLengthMS = readUINT32LE(in);
+	fread(&dro2hdr.iHardwareType, 1, 6, in);
+	if(dro2hdr.iCodemapLength >= 128) {
+		fprintf(stderr, "invalid setting %u for iCodemapLength!\n", (unsigned) dro2hdr.iCodemapLength);
+		return 2;
+	}
+	fread(dro2hdr.iCodemap, 1, dro2hdr.iCodemapLength, in);
+	dro2hdr_out = dro2hdr;
+	dro2hdr_out.iLengthPairs -= n;
+	dro2hdr_out.iLengthMS;
```
Comment (Malvineous) [inline droshrink.c]: This line does nothing now, did you mean to remove it entirely?

Comment (rofl0r) [reply droshrink.c]: fixed.

**Pull request merged**